apps/src/gamelab/GameLab.js

```diff
@@ -404,7 +403,8 @@ GameLab.prototype.init = function (config) {
     config.initialAnimationList : this.startAnimations;
   getStore().dispatch(setInitialAnimationList(initialAnimationList));
 
-  const loader = this.loadLibraries_().then(() => ReactDOM.render((
+  this.loadValidationCodeIfNeeded_();
+  const loader = this.studioApp_.loadLibraries_(this.level.helperLibraries).then(() => ReactDOM.render((
```

Comment:
Hmm... I know this isn't a change, but why does our initial render have to wait until the helper libraries are loaded, if we don't need them until we run the program?

Reply:
A couple of solutions come to mind.

1. The actual run-button-click could depend on an asynchronous getLibraries step, in a lazy-load sense, so that the first run would be slow but subsequent runs would be fast.  Then add a fire-and-forget preload step so that the first run is fast _unless_ they manage to hit run before the preload finishes.  I think this would work seamlessly on a share page.

2. We could actually disable the run button until the preload finishes, so we're doing the initial render immediately but not allowing the student to try running until the resources are available.  This might require some special handling on the share page.

Reply:
How do we do it for preloading the animation tab assets today?